kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/arbs.kt

```diff
@@ -104,10 +129,15 @@ fun <A, B> Arb<A>.distinctBy(selector: (A) -> B) = object : Arb<A>() {
  */
 fun <A, B : A> Arb<A>.merge(other: Gen<B>): Arb<A> = object : Arb<A>() {
    override fun edgecases(): List<A> = when (other) {
-      is Arb -> this@merge.edgecases().zip(other.edgecases()).flatMap { listOf(it.first, it.second) }
```

Comment:
this is wrong. these edgecases need to be concatenated instead of zipped. I've fixed this too.